allensdk/brain_observatory/filtered_running_speed/multi_stimulus_running_speed.py

```diff
@@ -0,0 +1,330 @@
+import numpy as np
+import pandas as pd
+from allensdk.brain_observatory.sync_dataset import Dataset as SyncDataset
+from allensdk.brain_observatory import sync_utilities
+import argschema
+import json
+
+from allensdk.brain_observatory.filtered_running_speed._schemas import (
+    MultiStimulusRunningSpeedInputParameters,
+    MultiStimulusRunningSpeedOutputParameters
+)
+
+from allensdk.brain_observatory.behavior.data_objects.\
+    running_speed.running_processing import (
+        get_running_df
+    )
+
+
+class MultiStimulusRunningSpeed(argschema.ArgSchemaParser):
+    default_schema = MultiStimulusRunningSpeedInputParameters
+    default_output_schema = MultiStimulusRunningSpeedOutputParameters
+
+    START_FRAME = 0
+
+    def _extract_dx_info(
+        self,
+        frame_times: np.ndarray,
+        start_index: int,
+        end_index: int,
+        pkl_path: str
+    ) -> pd.core.frame.DataFrame:
+        """
+        Extract all of the running speed data
+
+        Parameters
+        ----------
+        frame_times: numpy.ndarray
+            list of the vsync times
+        start_index: int
+            Index to the first frame of the stimulus
+        end_index: int
+           Index to the last frame of the stimulus
+        pkl_path: string
+            Path to the stimulus pickle file
+
+        Returns
+        -------
+        pd.DataFrame
+
+        Notes
+        -------
+            velocity pd.DataFrame:
+                columns:
+                    "velocity": computed running speed
+                    "net_rotation": dx in radians
+                    "frame_indexes": frame indexes into
+                        the full vsync times list
+
+            raw data pd.DataFrame:
+                Dataframe with an index of timestamps and the following
+                columns:
+                    "vsig": voltage signal from the encoder
+                    "vin": the theoretical maximum voltage that the encoder
+                        will reach prior to "wrapping". This should
+                        theoretically be 5V (after crossing
+                        5V goes to 0V, or vice versa). In
+                        practice the encoder does not always
+                        reach this value before wrapping, which can cause
+                        transient spikes in speed at the voltage "wraps".
+                    "frame_time": list of the vsync times
+                    "dx": angular change, computed during data collection
+                The raw data are provided so that the user may compute
+                their own speed from source, if desired.
+
+        """
+
+        stim_file = pd.read_pickle(pkl_path)
```

Comment:
I just want to make sure: reading the pickle file isn't time consuming, right? You do it in multiple places (each of the `get_frame_count` methods), and I want to make sure this isn't something we should be caching.

It's been so long since I touched one of the pickle files, I honestly can't remember how quick this is...

Reply:
It is not time consuming.